Assignment01/step-by-step-python.md

```diff
@@ -0,0 +1,191 @@
+
+## Step 1. Run the VehicleRegistration service
+
+1. Open the `python` folder in VS Code. This guide assumes VS Code, but feel free to use an editor or IDE you're
+   comfortable with.
+
+   > Throughout the assignment you can execute all steps in the same instance of editor or IDE window.
+
+2. Open a terminal window.
+
+   > You can do this by using the hotkey ``Ctrl-` `` (Windows) or ``Shift-Ctrl-` `` (macOS).
+
+3. Make sure the current folder is `python/VehicleRegistrationService`.
+
+4. Run the command `pip install -r requirements.txt`
+5. Start the service using `uvicorn vehicle_registration:app --port 6002`
+
+> If you receive an error here, please double-check whether or not you have installed all the
+> [prerequisites](../README.md#Prerequisites) for the workshop!
+
+Now you can test whether you can call the VehicleRegistrationService. You can do this using a browser, cURL or some
+other HTTP client. But there is a convenient way of testing RESTful APIs directly from VS Code (this uses the REST
+Client extension VS Code):
+
+1. Open the file `python/VehicleRegistrationService/test.http` in your editor. The request in this file simulates
+   retrieving the vehicle- and owner information for a certain license-number.
+
+2. Click on `Send request` in the file to send a request to the API:
+
+   ![REST client](img/rest-client.png)
+
+3. The response of the request will be shown in a separate window on the right. It should be a response with HTTP
+   status code `200 OK` and the body should contain some random vehicle and owner-information:
+
+   ```json
+   HTTP/1.1 200 
+   Connection: keep-alive
+   Content-Type: application/json
+   Date: Wed, 16 Jun 2021 19:39:05 GMT
+   Keep-Alive: timeout=60
+   Transfer-Encoding: chunked
+   
+   {
+       "vehicle_id": "KZ-49-VX",
+       "brand": "Toyota",
+       "model": "Rav 4",
+       "owner_name": "Angelena Fairbairn",
+       "owner_email": "angelena.fairbairn@outlook.com"
+   }
+   ```
+
+4. Check the logging in the terminal window. It should look like this:
+
+   ```console
+   > uvicorn vehicle_registration:app --port 6002
+   INFO:     Started server process [29384]
+   INFO:     Waiting for application startup.
+   INFO:     Application startup complete.
+   INFO:     Uvicorn running on http://127.0.0.1:6002 (Press CTRL+C to quit)
+   INFO:     127.0.0.1:64855 - "GET /vehicleinfo/KZ-49-VX HTTP/1.1" 200 OK
+   ```
+
+## Step 2. Run the FineCollection service
+
+1. Make sure the VehicleRegistrationService service is running (result of step 1).
+
+1. Open a **new** terminal window in VS Code.
+
+   > You can do this by using the hotkey (``Ctrl-` `` on Windows, ``Shift-Ctrl-` `` on macOS) or clicking on
+   > the `+` button in the terminal window title bar:
+   > ![](img/terminal-new.png)
+
+1. Make sure the current folder is `python/FineCollectionService`.
+
+1. Run the command `pip install -r requirements.txt`
+
+1. Start the service using `uvicorn fine_collection:app --port 6001`.
```

Comment:
This gives me

```console
Traceback (most recent call last):
  File "/usr/local/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/local/lib/python3.9/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/uvicorn/main.py", line 425, in main
    run(app, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/uvicorn/main.py", line 447, in run
    server.run()
  File "/usr/local/lib/python3.9/site-packages/uvicorn/server.py", line 68, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/base_events.py", line 642, in run_until_complete
    return future.result()
  File "/usr/local/lib/python3.9/site-packages/uvicorn/server.py", line 76, in serve
    config.load()
  File "/usr/local/lib/python3.9/site-packages/uvicorn/config.py", line 448, in load
    self.loaded_app = import_from_string(self.app)
  File "/usr/local/lib/python3.9/site-packages/uvicorn/importer.py", line 24, in import_from_string
    raise exc from None
  File "/usr/local/lib/python3.9/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/Cellar/python@3.9/3.9.7/Frameworks/Python.framework/Versions/3.9/lib/python3.9/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1030, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1007, in _find_and_load
  File "<frozen importlib._bootstrap>", line 986, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 680, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 850, in exec_module
  File "<frozen importlib._bootstrap>", line 228, in _call_with_frames_removed
  File "/Users/maarten/Junk/dapr-workshop-python/python/FineCollectionService/./fine_collection/__init__.py", line 3, in <module>
    from . import models, settings, services, clients
  File "/Users/maarten/Junk/dapr-workshop-python/python/FineCollectionService/./fine_collection/services.py", line 1, in <module>
    from . import clients, models, templates
  File "/Users/maarten/Junk/dapr-workshop-python/python/FineCollectionService/./fine_collection/templates/__init__.py", line 3, in <module>
    from jinja2 import Environment, FileSystemLoader, select_autoescape
ModuleNotFoundError: No module named 'jinja2'

```

Reply:
Woops! missing dependency. Fixed it in commit b081abe3fee10574e62dbe33b44709eaa8f5bb3f